src/main/java/edu/hm/hafner/coverage/parser/MetricsParser.java

```diff
@@ -0,0 +1,243 @@
+            final StartElement startElement, final String fileName)
+            throws XMLStreamException {
+        var sourceFilePath = Paths.get(getValueOf(startElement, NAME)).getFileName();
+        String sourceFilefileName;
+        if (sourceFilePath == null) {
+            sourceFilefileName = getValueOf(startElement, NAME);
+        }
+        else {
+            sourceFilefileName = sourceFilePath.toString();
+        }
+        var fileNode = packageNode.findOrCreateFileNode(sourceFilefileName,
+                internPath(getValueOf(startElement, NAME)));
+
+        while (reader.hasNext()) {
+            XMLEvent event = reader.nextEvent();
+
+            if (event.isStartElement()) {
+                var nextElement = event.asStartElement();
+                if (CLASS.equals(nextElement.getName())) {
+                    readClass(reader, fileNode, nextElement, fileName, packageNode);
+                }
+                else if (METRIC.equals(nextElement.getName())) {
+                    readValueCounter(fileNode, nextElement);
+                }
+            }
+            else if (event.isEndElement()) {
+                var endElement = event.asEndElement();
+                if (FILE.equals(endElement.getName())) {
+                    return fileNode;
+                }
+            }
+        }
+        throw createEofException(fileName);
+    }
+
+    @CanIgnoreReturnValue
+    private Node readMethod(final XMLEventReader reader, final ClassNode classNode,
+            final StartElement startElement, final String fileName) throws XMLStreamException {
+        String methodName = getValueOf(startElement, NAME) + "#" + getValueOf(startElement, BEGIN_LINE);
+
+        MethodNode methodNode = createMethod(startElement, methodName);
+        classNode.addChild(methodNode);
+
+        while (reader.hasNext()) {
+            XMLEvent event = reader.nextEvent();
+
+            if (event.isStartElement()) {
+                var nextElement = event.asStartElement();
+                if (METRIC.equals(nextElement.getName())) {
+                    readValueCounter(methodNode, nextElement);
+                }
+            }
+            else if (event.isEndElement()) {
+                var endElement = event.asEndElement();
+                if (METHOD.equals(endElement.getName())) {
+                    return methodNode;
+                }
+            }
+        }
+        throw createEofException(fileName);
+    }
+
+    private MethodNode createMethod(final StartElement startElement, final String methodName) {
+        return new MethodNode(methodName, "", parseInteger(getValueOf(startElement, BEGIN_LINE)));
+    }
+
+    private void readValueCounter(final Node node, final StartElement startElement) {
+        String currentType = getValueOf(startElement, NAME);
+        int value = parseInteger(getValueOf(startElement, VALUE));
+        if (CYCLOMATIC_COMPLEXITY.equals(currentType)) {
```

Comment:
Lines 230-241 are ok for now but in the future it makes sense to create the metric directly in the style
```
 node.addValue(new MetricValue(value, Metric.of(currentType))
```
Or what do you think?

Reply:
Yes, as soon as arbitrary Metrics are possible in the Coverage Model and Plugin, it should be created like this.